packages/smart-contracts/src/contracts/ERC20EscrowToPayV1.sol

```diff
@@ -0,0 +1,385 @@
+
+    /**
+     * @notice Emitted when a tokentimelock period is completed successfully.
+     * @param paymentReference Reference of the payment related.
+     */
+    event LockPeriodEnded(bytes indexed paymentReference);
+    
+    constructor(address payable _paymentProxyAddress) {
+        owner = payable(msg.sender);
+        paymentProxy = IERC20FeeProxy(_paymentProxyAddress);
+    }
+
+    /// @notice recieve function reverts and returns the funds to the sender.
+    receive() external payable {
+        revert("not payable receive");
+    }
+
+    /** 
+     * @notice Stores the invoice details in struct, then transfers the funds to this Escrow contract.
+     * @param _paymentRef Reference of the payment related.
+     * @param _paymentToken Address of the ERC20 token to deposit.
+     * @param _payee Transfer recipient.
+     * @param _amount Amount to transfer.
+     * @param _feeAmount The amount of fee to pay.
+     * @param _feeAddress The fee recipient.
+     * @dev The fees amount is transfered from the payer's account.
+     * @return success if transaction successeded. 
+     */
+    function openEscrow(
+        bytes memory _paymentRef,
+        IERC20 _paymentToken,
+        uint256 _amount,
+        address _payee,
+        uint256 _feeAmount,
+        address _feeAddress
+    ) 
+        external
+        payable
+        returns (bool success)
+    {
+        require(invoiceMapping[_paymentRef].amount == 0, 
+            "ERC20EscrowToPayV1: This invoice already exists, is this the correct paymentReference?"
+        ); 
+        
+        uint256 sixMonths = 15778458;
+        // claimDate is used to set a timestamp on when the payee can execute closeEscrow.
+        uint256 _claimDate = block.timestamp + sixMonths;
+        
+        invoiceMapping[_paymentRef] = Invoice(
+            _paymentToken, 
+            _amount,
+            _payee,
+            msg.sender,
+            _feeAmount,
+            _feeAddress,
+            _claimDate
+        );
+        
+        require(_deposit(_paymentRef),"OpenEscrow: Call to _deposit failed!");
+
+        emit OpenEscrow(_paymentRef);
+        return true;
+    }
+    
+    /**
+     * @notice Closes an open escrow and pays the invoice request to it's issuer.
+     * @param _paymentRef Reference of the related Invoice.
+     * @dev The fees is paid to the fee address at this point.
+     */
+    function closeEscrow(bytes memory _paymentRef) external {
+        require(invoiceMapping[_paymentRef].amount != 0, "ERC20EscrowToPay: Payment reference does not exist");
+        require(msg.sender == invoiceMapping[_paymentRef].payer ||
+            (msg.sender == invoiceMapping[_paymentRef].payee  &&
+            invoiceMapping[_paymentRef].claimDate <= block.timestamp),
+            "ERC20EscrowToPay: Only the payer or payee can excecute this call!"
+        );
+        require(_withdraw(_paymentRef, invoiceMapping[_paymentRef].payee), "CloseEscrow: Withdraw failed!");
```

Comment:
I may ignore the good practices here but in other `require` messages, you start with the name of the contract, and here with `CloseEscrow`, it's a bit unconsistent.

Reply:
```suggestion
        require(_withdraw(_paymentRef, invoiceMapping[_paymentRef].payee));
```

I think a string description is redundant here, because `_withdraw(_paymentRef, invoiceMapping[_paymentRef].payee)` will throw `WithdrawFailed("Withdraw FAILED")`  if transaction fails.